bindings/rust-examples/session-resumption/README.md

```diff
@@ -0,0 +1,13 @@
+This example demonstrates the s2n-tls [session resumption](https://aws.github.io/s2n-tls/usage-guide/ch11-resumption.html) feature. First, the client connects to the server and retrieves a session ticket. Then, the client connects to the server again, but uses the retrieved session ticket to resume the previous session.
```

Comment:
Yay for more examples 🕺 !

Two broad points
1. The "ApplicationContext" demonstration feels a bit shoe-horned into this example. There is no need in this scenario to lock session tickets to a specific IP address, and generally client implementers _should not_ do that unless they have specific information about distributed session resumption not being supported. I worry that someone is going to look at this and say "oh, session resumption should only happen with the same ip address". I think we should either
    - only focus on session resumption in this example
    - expand the example to use two servers with different STEKs, and demonstrate the success (ip specific resumption) and failure (attempt distributed resumption) modes.
2. If we choose to go the latter route, I'd vote for using turmoil in the example. This allows all of the clients/servers to be simulated in the single process (no need for 3 terminal windows) and also allows the example to be explicitly testable. You can see an example of the turmoil stuff [here](https://github.com/jmayclin/s2n-tls/blob/psk-bindings/bindings/rust-examples/external-psk/src/lib.rs)

Note that if your goal is to just have an example of using the `set_application_context` method, a [doc comment](https://doc.rust-lang.org/rust-by-example/meta/doc.html#doc-comments) might be sufficient?